client/client.go

```diff
@@ -515,6 +515,68 @@ func (r *NotaryRepository) GetChangelist() (changelist.Changelist, error) {
 	return cl, nil
 }
 
+// GetRepoKeyIDs returns a map of associations between this repo's roles and its key IDs
+func (r *NotaryRepository) GetRepoKeyIDs() (map[string][]string, error) {
+	// Update to latest repo state
+	_, err := r.Update(false)
+	if err != nil {
+		return nil, err
+	}
+
+	keys := make(map[string][]string)
+
+	// Add all keys for this repo
+	for keyPath, role := range r.CryptoService.ListAllKeys() {
+		gun := ""
+		// Root keys are shared across repos, no need to find gun
+		if role != data.CanonicalRootRole {
+			dirPath := filepath.Dir(keyPath)
+			if dirPath != "." { // no gun
+				gun = dirPath
+			}
+			if gun != r.gun {
+				continue
+			}
+		}
+		// Chomp off the leading role from the key ID, and add to our map
+		keys[role] = append(keys[role], filepath.Base(keyPath))
+	}
+	return keys, nil
+}
+
+// GetLatestSignatureKeyIDs returns a map of associations between this repo's roles
+// and the latest key IDs used for signing that role's metadata
+func (r *NotaryRepository) GetLatestSignatureKeyIDs() (map[string][]string, error) {
+	// Update to latest repo state
+	_, err := r.Update(false)
+	if err != nil {
+		return nil, err
+	}
+
+	keys := make(map[string][]string)
+
+	rootSignatures := r.tufRepo.Root.Signatures
+	for _, sig := range rootSignatures {
+		keys[data.CanonicalRootRole] = append(keys[data.CanonicalRootRole], sig.KeyID)
+	}
+	// Traverse into the targets map, this includes delegations
+	targets := r.tufRepo.Targets
```

Comment:
I think this may only get the first level delegations?  Delegations can be nested (so "targets/a" will have the keys for "targets/a/b" for instance.  So I think you may have to walk down the tree of targets to get all the key IDs.